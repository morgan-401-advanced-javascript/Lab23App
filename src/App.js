import React from "react";
import { BrowserRouter, Route, Link } from "react-router-dom";
import Home from "./home.js";
import Header from "./header.js";
import Api from "./api.js";
import Sockets from "./sockets";
import bootstrap from "bootstrap";

function App() {
  return (
    <BrowserRouter>
      <div className="App">
        <h1>LAB 23!</h1>
        <Header className="nav">
          <Link className="btn btn-info" to="/">
            Home
          </Link>
          <Link className="btn btn-info" to="/sockets">
            Check Weather
          </Link>
          <Link className="btn btn-info" to="/api">
            Get Wise
          </Link>
        </Header>
        <Route exact path="/" component={Home} />
        <Route exact path="/sockets" component={Sockets} />
        <Route exact path="/api" component={Api} />
      </div>
    </BrowserRouter>
  );
}

export default App;
