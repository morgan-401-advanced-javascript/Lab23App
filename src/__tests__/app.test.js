import React from 'react';
import App from '../components/App';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({adapter:new Adapter()});

describe('', () => {
    it('', () => {
        expect(true).toBeTruthy();
    });
});

describe('App component', () => {
    it('renders correctly', () => {
        const page = renderer.create(<App />).toJSON();
        expect(page).toMatchSnapshot();
    });
});

describe('App Enzyme Testing', () => {
    it('has a  Heading', () => {
        let component = mount(<App />);
        expect(component.find('div.App h1').text()).toBe(
            'LAB 23!',
        );
    })});