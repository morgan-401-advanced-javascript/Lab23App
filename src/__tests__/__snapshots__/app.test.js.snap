// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    LAB 23!
  </h1>
  <div
    className="nav nav-pills mb-3"
    id="pills-tab"
    role="tablist"
  >
    <div>
      <a
        className="btn btn-info"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="btn btn-info"
        href="/sockets"
        onClick={[Function]}
      >
        Check Weather
      </a>
      <a
        className="btn btn-info"
        href="/api"
        onClick={[Function]}
      >
        Get Wise
      </a>
    </div>
  </div>
  <div>
    <h3>
      Welcome!
    </h3>
    <p>
      Feel free to check the weather or receive wisdom.
    </p>
  </div>
</div>
`;
